### ISSUE TITLE
Lazy loading of partials

### ISSUE THREAD

**Issue by warpech** [Starcounter/Starcounter#2894]
@careri asked how would we do lazy loading of partials. Meaning, a partial (including template and data!) should be loaded only when I scroll down to a DIV that holds it.

I think this is extension of a well known method of lazy loading of images. This would be useful for charts, data grids, etc.

I think we would have 2 properties in JSON:

```json
{
 "myChart": {},
 "myChartVisible": false
}
```

On the client side, there would be a Web Component that sets the property `myChartVisible: true`. In result, server would populate `myChart` with the actual graph partial using `Self.GET`.

Let's test it out some day!

**Comment by tomalec** [comment]
Have you seen Eric Bidelman's tweet? https://twitter.com/ebidel/status/641278429032087552

**Comment by warpech** [comment]
Yeah! And also I am looking forward to their Google's Carbon lazy loading element, announced here: https://youtu.be/jVn8tlnwAEs?t=20m8s. I guess it will be very similar stuff
